pkg/scale/decode.go

```diff
@@ -426,6 +428,34 @@ func (ds *decodeState) decodeArray(dstv reflect.Value) (err error) {
 	return
 }
 
+func (ds *decodeState) decodeMap(dstv reflect.Value) (err error) {
```

Comment:
nit rename `dstv` to `destination`

Reply:
It is dstv everywhere in that file, there is no need to change this